proto/process/agent.proto

```diff
@@ -488,13 +489,18 @@ message Connection {
 
 	// dns stats based on domain queried, the key corresponds to an index into the `domains` field
 	map<int32, DNSStats> dnsStatsByDomain = 34;
+
+	// httpKey corresponds to an index into the 'httpKeys' field
+	int32 httpKey = 35;
 }
 
 message Connections {
   repeated Connection conns = 1;
   map<string, DNSEntry> dns  = 2;
   ConnectionsTelemetry telemetry = 3;
   repeated string domains = 4;
+  repeated HTTPKey httpKeys = 5; // keys for the http map below
+  map<int32, HTTPStatsByPath> http = 6;
```

Comment:
There is lots of indirection here. I suggest:

* change the type of `http` to `releated HTTPStatsByPath`
* include `source/dest` information in `HTTPStatsByPath`
* include an index into the `http` field in each `Connection`